autoemulate/experimental/model_selection.py

```diff
@@ -24,31 +25,37 @@
 logger = logging.getLogger("autoemulate")
 
 
+def r2_metric() -> type[torchmetrics.Metric]:
+    """Returns a torchmetrics.R2Score metric."""
+    return torchmetrics.R2Score
+
+
+def rmse_metric() -> partial[torchmetrics.Metric]:
+    """Returns a torchmetrics.MeanSquaredError metric with squared=False."""
+    return partial(torchmetrics.MeanSquaredError, squared=False)
+
+
 def _update(
     y_true: InputLike,
     y_pred: OutputLike,
     metric: torchmetrics.Metric,
 ):
     # handle types
     if isinstance(y_pred, TensorLike):
-        metric.update(y_true, y_pred)
+        metric.to(y_pred.device)
+        metric.update(y_pred, y_true)
     elif isinstance(y_pred, DistributionLike):
-        metric.update(y_true, y_pred.mean)
-    elif (
-        isinstance(y_pred, tuple)
-        and len(y_pred) == 2
-        and all(isinstance(item, TensorLike) for item in y_pred)
-    ):
-        metric.update(y_true, y_pred)
+        metric.to(y_pred.mean.device)
+        metric.update(y_pred.mean, y_true)
     else:
         raise ValueError(f"Metric not implmented for {type(y_pred)}")
 
 
 def evaluate(
-    y_true: InputLike,
     y_pred: OutputLike,
-    metric: type[torchmetrics.Metric],
-    device: DeviceLike,
+    y_true: InputLike,
+    metric: type[torchmetrics.Metric]
+    | partial[torchmetrics.Metric] = torchmetrics.R2Score,
```

Comment:
This update and elsewhere in this module is since the `torchmetrics` API expects predictions first and target values second in postional args.